### PR TITLE
Encapsulate seq state: Take 2

### PR DESCRIPTION
**Pull request by catkins**
Same changes as #27 but rebased onto `main` now that #28 has been merged

Re-arranging pattern state and settings into same object tree. May make things a bit more Object Oriented, such that Pattern can handle some logic itself. No functional changes, just shuffling around some code (pure [refactoring](https://en.wikipedia.org/wiki/Code_refactoring))

Some of the motivation: http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#i2-avoid-non-const-global-variables

Note: Best reviewed in split mode, and hiding whitespace diffs.
https://github.com/okyeron/OMX-27/pull/32/files?diff=split&w=1

---

There's plenty more that could be done to move actual logic into the sequencer class, but I think that's best left to future PRs.

### CODE REVIEW

File: OMX-27-firmware/OMX-27-firmware.ino [hunk 1]
```diff
@@ -295,13 +298,6 @@ void setup() {
 	lastProcessTime = micros();
 	resetClocks();
 
-	nextStepTime = micros();
```
Comment (catkins) [inline OMX-27-firmware/OMX-27-firmware.ino:298]: this stuff was moved into the sequencer `defaultSequencer()` function

File: OMX-27-firmware/sequencer.h [hunk 2]
```diff
@@ -157,15 +195,22 @@ int loopCount[NUM_PATTERNS][NUM_STEPS] = {
 	{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}
 };
 
-const char* trigConditions[36] = {"1:1","1:2","2:2","1:3","2:3","3:3","1:4","2:4","3:4","4:4","1:5","2:5","3:5","4:5","5:5","1:6","2:6","3:6","4:6","5:6","6:6","1:7","2:7","3:7","4:7","5:7","6:7","7:7","1:8","2:8","3:8","4:8","5:8","6:8","7:8","8:8"};
-int ABcondition = 0;
```
Comment (catkins) [inline OMX-27-firmware/sequencer.h:161]: this was unused

Comment (okyeron) [reply OMX-27-firmware/sequencer.h:161]: `trigConditions[36]` was there for legendText in `SUBMODE_NOTESEL2` - line 844 in your revisions

Comment (okyeron) [reply OMX-27-firmware/sequencer.h:161]: Oops - sorry, didnt see the replaced version

**Comment by okyeron** [comment]
sequencer.h line 52
This needs to be 6 for 64 steps.

I posted a review for that above, but can't figure out how to link back to it.

**Pull request merged**